File: eAutoShop/eAutoShop.Model/SearchObjects/AppointmentSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eAutoShop.Model.SearchObjects
{
    public class AppointmentSearchObject : BaseSearchObject
    {
        public string? CustomerUsername { get; set; }

        public string? EmployeeUsername { get; set; }

        public string? State { get; set; }

        public string? Type { get; set; }

        public double? MinTotalAmount { get; set; }

        public double? MaxTotalAmount { get; set; }

        public bool? HasOrder { get; set; }

        public DateTime? MinCreatedDate { get; set; }

        public DateTime? MaxCreatedDate { get; set; }

        public DateTime? MinReservationDate { get; set; }

        public DateTime? MaxReservationDate { get; set; }

        public DateTime? MinCompletionDate { get; set; }

        public DateTime? MaxCompletionDate { get; set; }
    }
}

File: eAutoShop/eAutoShop.UI/eautoshop_desktop/lib/models/appointment/appointment_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment_search_object.g.dart';

@JsonSerializable(includeIfNull: false)
class AppointmentSearchObject {
  final String? customerUsername;
  final String? employeeUsername;
  final String? state;
  final String? type;
  final double? minTotalAmount;
  final double? maxTotalAmount;
  final bool? hasOrder;
  final DateTime? minCreatedDate;
  final DateTime? maxCreatedDate;
  final DateTime? minReservationDate;
  final DateTime? maxReservationDate;
  final DateTime? minCompletionDate;
  final DateTime? maxCompletionDate;

  const AppointmentSearchObject({
    this.customerUsername,
    this.employeeUsername,
    this.state,
    this.type,
    this.minTotalAmount,
    this.maxTotalAmount,
    this.hasOrder,
    this.minCreatedDate,
    this.maxCreatedDate,
    this.minReservationDate,
    this.maxReservationDate,
    this.minCompletionDate,
    this.maxCompletionDate,
  });

  factory AppointmentSearchObject.fromJson(Map<String, dynamic> json) =>
      _$AppointmentSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentSearchObjectToJson(this);
}
